### ISSUE TITLE
Organic Maps not available in Android Auto

### ISSUE THREAD

**Issue by chrisdebian** [organicmaps/organicmaps#7066]
Hi, all.

I'm using 2023.12.19-15-Google-beta on an Android 14 Pixel 6.

Organic Maps is not available on my Android Auto head unit, and it does not appear on my Android Auto phone settings, as an app that I can display on Android Auto (see screenshot). I'm unable to test Organic Maps on Android Auto, because I have no way of opening it.

![Screenshot_20240103-170642](https://github.com/organicmaps/organicmaps/assets/17935223/fea61f24-db75-4633-ba95-a201fae9be3c)

Can anyone help?

Thanks,

Chris

**Comment by Jean-BaptisteC** [comment]
It's not the right beta to test Android Auto, please send mail to hello@organicmaps.app to get access to test Android Auto beta

**Comment by chrisdebian** [comment]
My mistake, I've requested access to this Beta build.

Thanks,

Chris

**Comment by chrisdebian** [comment]
Closing, as was using the wrong beta.

**Comment by chrisdebian** [comment]
Hi, all.

I have the latest beta version installed from Google Play, that includes Android Auto. Unfortunately this problem still exists, so I'm unable to test on a head unit, because the app doesn't appear on the display.

Thanks,

Chris

**Comment by biodranik** [comment]
@chrisdebian what is the version of your beta?

**Comment by chrisdebian** [comment]
Hi,

It's 2024.01.09-5-Google-beta

Thanks,

Chris

**Comment by kaibagley** [comment]
I am also having this issue, but potentially for a different reason.

Phone: Pixel 7 Pro
OS: GrapheneOS / AOSP 14
Google services installed, no network access

Using the AA beta linked after emailing OM, the current Gplay version, the Fdroid version, and the GitHub version, OM won't appear in the AA launcher for me.

**Comment by Jean-BaptisteC** [comment]
Did you have the latest version of AA?
https://play.google.com/store/apps/details?id=com.google.android.projection.gearhead

**Comment by kaibagley** [comment]
Yes:
- Google Play: 11.0.635014-release (110635014)
- GrapheneOS Apps: 11.0.635014-release

**Comment by Jean-BaptisteC** [comment]
I guess problem is caused because it's a beta version, can you test with latest release?
To check is app can be use on Android Auto
Go to settings -> Android Auto settings -> Go to the bottom and touch version and check if Organic Maps is in app list

**Comment by kaibagley** [comment]
In Android Auto, the "Version and permission info" section says:

> Packages:
> ...
> Organic Maps: 2024.01.06-5-Web
> ...

I've tried the beta version too, but this should be the github version downloaded using Obtainium

**Comment by chrisdebian** [comment]
I have the latest AA, and it is reporting the following version of Organic Maps, when I click on "Version and Permission Information":

Packages:

Organic Maps (Beta): 2024.01.09-5-Google-beta

Permissions:
phone: on
contacts: on
location: on
sms: on
microphone: on
calendar: off
call_log: on
nearby_devices: on

Thanks,

Chris

**Comment by AndrewShkrob** [comment]
What version of Android Auto app do you have?
Is it the last one?
This problem may be related to this issue: https://github.com/organicmaps/organicmaps/issues/6105

**Comment by chrisdebian** [comment]
Android Auto: 10.9.634644-release
Organic Maps (Beta): 2024.01.09-5-Google-beta

**Comment by AndrewShkrob** [comment]
Ok, I rechecked. Waze still has `minCarApiLevel = 5`. It's not related to #6105.

@chrisdebian, perhaps you should wait for some "magic" to happen inside the Android Auto app, and it might start recognizing Organic Maps. Alternatively, you can try fully reinstalling AA and OM apps or rebooting your device. 😄

**Comment by chrisdebian** [comment]
@AndrewShkrob Thank you for trying.

Chris

**Comment by knuxyl** [comment]
I just reinstalled/updated to the latest LineageOS for MicroG on my Pixel 5 and now Organic Maps doesn't show in Android Auto anymore and it did yesterday. I installed AA4MG (https://github.com/sn-00-x/aa4mg) and then Android Auto. I've been using release builds from Fdroid, not beta. Nothing has changed about my installation from yesterday except I updated to the latest LineageOS, which I doubt is the problem. This might not be related or just user error somehow, just thought I'd share what I'm experiencing.

Android Auto: 11.1.640234-release (Installed as root from Aurora Store, mimicks Play Store installation)
Organic Maps: 2023.09.22-27-FDroid
LineageOS: 20-20240113-microg-redfin (Android 13)

**Comment by Jean-BaptisteC** [comment]
Latest version on F-Droid doesn't have Android auto

**Comment by knuxyl** [comment]
> Latest version on F-Droid doesn't have Android auto

Yeah I realized that, but the latest from github still does not show in the auto launcher, but does show in the version info on the auto app. It only shows when I use an lsposed module with magisk.

chrisdebian seems to have the exact same problem, with it showing under packages but not available in the launcher.

**Comment by michaelblyons** [comment]
> I think it may be possible it's not showing due to not being installed from the Play Store, although I have VLC from Fdroid that works fine. There may be a difference since this uses maps api, I don't know.

While the docs say that you can't use _any_ AA apps unless they're installed from Play, anecdotal evidence suggests that this is not enforced for media and messaging. Navigation is blocked, though. I don't know if there are other APIs and what their behavior is.

**Comment by AndrewShkrob** [comment]
> Android Auto: 10.9.634644-release 
> Organic Maps (Beta): 2024.01.09-5-Google-beta

@chrisdebian I've noticed that your beta from play store has "-beta" suffix. That's incorrect. Beta versions from the store shouldn't contain it. It's probably from from the AppTester app, not from play store. Beta versions from the AppTester app are not supported. Please, check that you really has the beta (or release) version from play store.

**Comment by chrisdebian** [comment]
Ok, so the beta isn't a beta, and the AppTester isn't supported. Isn't the point of apptesting, to enable feedback? I'll withdraw from any betas, and just wait for the functionality to magically arrive. Confused.

Chris

**Comment by chrisdebian** [comment]
Beta removed, stock app installed from Play. Organic Maps seems to be available to Android Auto, now.

![Screenshot_20240201-192818](https://github.com/organicmaps/organicmaps/assets/17935223/12ffe0e2-fb83-40ab-a805-9e54f6453565)

**Comment by AndrewShkrob** [comment]
> Beta removed, stock app installed from Play. Organic Maps seems to be available to Android Auto, now.

Great! I'm closing the issue :)

**Comment by kaibagley** [comment]
I'm still having this issue. Should I open a new issue?

**Comment by AndrewShkrob** [comment]
> I'm still having this issue. Should I open a new issue?

@kaibagley You wrote that you have Organic Maps 2024.01.06-5-Web. Please download the version from Google Play. This is the only version that officially supports Android Auto.

**Comment by kaibagley** [comment]
> I am also having this issue, but potentially for a different reason.
> 
> Phone: Pixel 7 Pro OS: GrapheneOS / AOSP 14 Google services installed, no network access
> 
> Using the AA beta linked after emailing OM, the current Gplay version, the Fdroid version, and the GitHub version, OM won't appear in the AA launcher for me.

I have used all of these versions. The only thing I can personally think of is that Google Play Services is required since I downloaded it through Aurora store (version 2024.01.09-6-Google, same as from gplay)

**Comment by Airyzz** [comment]
@kaibagley There is a security limitation making it so apps installed from sources other than the google play store do not work with android auto. Even if the APK you install is the exact some one that is released on google play, it will not work unless actually installed via google play.

See: https://github.com/ev-map/EVMap/issues/319

**Comment by Airyzz** [comment]
It was my understanding that the google play store requirement was less strictly enforced for media apps, So that may explain why a lot of those apps work. I tried to install Waze from Aurora store and much to my surprise, it did indeed show up with android auto, as well as if I just installed a waze apk from file browser. 

Curious to understand the mechanism behind this. Though, since Waze is owned by google, I wouldnt be suprised if waze gets special treatment. Maybe there is an allow list of apps which work regardless of install method
